### PR TITLE
chore: update contributing.md and PR template

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+    - docs(console): Update docstrings
+    - feat(doc): Handle nested reexports
+
+  Examples of bad title:
+    - fix #7123
+    - update docs
+    - fix bugs
+
+2. Ensure there is a related issue and it is referenced in the PR text.
+3. Ensure There are tests that cover the changes.
```
Comment (lucacasonato) [inline .github/PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
3. Ensure there are tests that cover the changes.
```

**Review by lucacasonato — APPROVED** [review]
LGTM. I hope people will actually read this :-D

**Review by ry — APPROVED** [review]

File: docs/contributing.md [hunk 2]
```diff
@@ -22,11 +25,24 @@ Instructions on how to build from source can be found
...
-2. There are tests that cover the changes.
-3. Ensure `cargo test` passes.
-4. Format your code with `./tools/format.py`.
-5. Make sure `./tools/lint.py` passes.
+1. Give the PR a descriptive title.
+
+Examples of good PR title:
+
+- fix(std/http): Fix race condition in server
+- docs(console): Update docstrings - feat(doc): Handle nested reexports
```
Comment (caspervonb) [inline docs/contributing.md:33]:
```suggestion
- docs(console): Update docstrings
- feat(doc): Handle nested reexports
```

**Comment by caspervonb** [comment]
Closes https://github.com/denoland/deno/issues/6843

**Review by trivikr — APPROVED** [review]

**Review by caspervonb — APPROVED** [review]
LGTM

**Pull request merged**